model/status_board/v1/application_dependency_type.model

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Definition of a Status Board application dependency.
+class ApplicationDependency {
+  // Object creation timestamp.
+  CreatedAt Date
+
+  // Object modification timestamp.
+  UpdatedAt Date
+
+  // Object deletion timestamp.
+  DeletedAt Date
+
+  // The name of the application. 
+  Name String
+
+  // Miscellaneous metadata about the application.
+  Metadata Interface
+
+  // The type of application dependency, e.g. soft or hard.
+  Type String
+
+  // The Service associated with the dependency.
+  ServiceId String
```

Comment:
In general it is better to have references to other objects implemented using the types corresponding to those objects. Here you could do this:

```
// The Service associated with the dependency:
Service Service
```

The JSON representation of that would then be something like this:

```json
{
  "kind": "ApplicationDependency",
  "id": "123",
  "href": "/api/status_board/v1/application_dependencies/123",
  "service": {
    "kind": "ServiceLink",
    "id": "456",
    "href": "/api/status_board/v1/services/456"
  },
  ...
}
```

Note how the `kind` attribute is `ServiceLink` instead of `Service`. That is what indicates that this is a reference to an object and not the object itself. Tools can then use this information to automatically retrieve the complete object if needed by just taking the value of the `href`  attribute and sending a `GET` request.

This is just a recommendation. You will find existing services that follow it and some that don't.

Same for the reference to the application below.

Reply:
Ok, updated.